File: app/src/main/java/com/legue/axel/lolsummonertool/database/model/match/TeamStat.java
package com.legue.axel.lolsummonertool.database.model.match;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.ForeignKey;
import android.arch.persistence.room.Ignore;
import android.arch.persistence.room.PrimaryKey;

import static android.arch.persistence.room.ForeignKey.CASCADE;

@Entity(
        tableName = "teamStats",
        foreignKeys = @ForeignKey(
                entity = Match.class,
                parentColumns = "gameId",
                childColumns = "matchId",
                onDelete = CASCADE)
)
public class TeamStat {

    @PrimaryKey(autoGenerate = true)
    public int id;
    public boolean firstDragon;
    public boolean firstInhibitor;
    public int baronKills;
    public boolean firstRiftHerald;
    public boolean firstBaron;
    public int riftHeraldKills;
    public boolean firstBlood;
    public int teamId; // 100 for blue side - 200 for red side
    public boolean firstTower;
    public int vilemawKills;
    public int inhibitorKills;
    public int towerKills;
    public int dominionVictoryScore;
    public String win; // String indicating whether or not the team won. There are only two values visibile in public Match history. (Legal values: Fail, Win)
    public int dragonKills;
    @ColumnInfo(index = true)
    public int matchId;


    @Ignore
    public TeamStat() {
    }

    public TeamStat(int id, boolean firstDragon, boolean firstInhibitor, int baronKills,
                    boolean firstRiftHerald, boolean firstBaron, int riftHeraldKills,
                    boolean firstBlood, int teamId, boolean firstTower, int vilemawKills,
                    int inhibitorKills, int towerKills, int dominionVictoryScore, String win,
                    int dragonKills, int matchId) {
        this.id = id;
        this.firstDragon = firstDragon;
        this.firstInhibitor = firstInhibitor;
        this.baronKills = baronKills;
        this.firstRiftHerald = firstRiftHerald;
        this.firstBaron = firstBaron;
        this.riftHeraldKills = riftHeraldKills;
        this.firstBlood = firstBlood;
        this.teamId = teamId;
        this.firstTower = firstTower;
        this.vilemawKills = vilemawKills;
        this.inhibitorKills = inhibitorKills;
        this.towerKills = towerKills;
        this.dominionVictoryScore = dominionVictoryScore;
        this.win = win;
        this.dragonKills = dragonKills;
        this.matchId = matchId;
    }
}

File: app/src/main/java/com/legue/axel/lolsummonertool/database/model/match/TeamStat.kt
package com.legue.axel.lolsummonertool.database.model.match

import androidx.annotation.NonNull
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.ForeignKey.CASCADE
import androidx.room.PrimaryKey

@Entity(tableName = "teamStats", foreignKeys = [ForeignKey(entity = Match::class, parentColumns = ["gameId"], childColumns = ["matchId"], onDelete = CASCADE)])
data class TeamStat(
        @PrimaryKey(autoGenerate = true)
        @NonNull
        var id: Int,
        var firstDragon: Boolean? = null,
        var firstInhibitor: Boolean? = null,
        var baronKills: Int? = null,
        var firstRiftHerald: Boolean? = null,
        var firstBaron: Boolean? = null,
        var riftHeraldKills: Int? = null,
        var firstBlood: Boolean? = null,
        var teamId: Int? = null, // 100 for blue side - 200 for red side
        var firstTower: Boolean? = null,
        var vilemawKills: Int? = null,
        var inhibitorKills: Int? = null,
        var towerKills: Int? = null,
        var dominionVictoryScore: Int? = null,
        var win: String? = null, // String indicating whether or not the team won. There are only two values visibile in public Match history. (Legal values: Fail, Win)
        var dragonKills: Int? = null,
        @ColumnInfo(index = true)
        var matchId: Long? = null
)